content/chainguard/chainguard-images/getting-started/wordpress.md

```diff
@@ -0,0 +1,305 @@
+
+This demo includes a theme ([cue](https://wordpress.org/themes/cue/), a simple blogging theme) and a plugin ([imsanity](https://wordpress.org/plugins/imsanity/), a popular plugin used to resize images) to demonstrate how to include custom content in the image.
+
+Navigate to the `03-distroless` directory to follow along. This is how the Dockerfile included in this directory looks like:
+
+```Dockerfile
+FROM cgr.dev/chainguard/wordpress:latest-dev as builder
+#trigger wp-config.php creation
+ENV WORDPRESS_DB_HOST=foo
+
+#copy wp-content folder
+COPY ./wp-content /usr/src/wordpress/wp-content
+
+#run entrypoint script
+RUN /usr/local/bin/docker-entrypoint.sh php-fpm --version
+
+FROM cgr.dev/chainguard/wordpress:latest
+
+COPY --from=builder --chown=php:php /var/www/html /var/www/html
+```
+
+Notice that we're copying the contents of the local `wp-content` folder to the `/usr/src/wordpress` folder in the container. This is the location of the WordPress source files. These will be copied to the document root by the entrypoint script that is executed right afterward. At the `builder` stage, we're also setting up a single environment variable to trigger the creation of the `wp-config.php` file that relies on a set of environment variables to configure database access.
+
+In the `docker-compose.yaml` file, we reference the custom Dockerfile:
+
+```yaml
+services:
+  app:
+    image: wordpress-local-distroless
+    build:
+      context: .
+      dockerfile: Dockerfile
+    restart: unless-stopped
+    environment:
+      WORDPRESS_DB_HOST: mariadb
+      WORDPRESS_DB_USER: $WORDPRESS_DB_USER
+      WORDPRESS_DB_PASSWORD: $WORDPRESS_DB_PASSWORD
+      WORDPRESS_DB_NAME: $WORDPRESS_DB_NAME
+    volumes:
+      - document-root:/var/www/html
+
+  nginx:
+    image: cgr.dev/chainguard/nginx
+    restart: unless-stopped
+    ports:
+      - 8000:8080
+    volumes:
+      - document-root:/var/www/html
+      - ./nginx.conf:/etc/nginx/nginx.conf
+
+  mariadb:
+    image: cgr.dev/chainguard/mariadb
+    restart: unless-stopped
+    environment:
+      MARIADB_ALLOW_EMPTY_ROOT_PASSWORD: 1
+      MARIADB_USER: $WORDPRESS_DB_USER
+      MARIADB_PASSWORD: $WORDPRESS_DB_PASSWORD
+      MARIADB_DATABASE: $WORDPRESS_DB_NAME
+    ports:
+      - 3306:3306
+
+volumes:
+  document-root:
+
+```
+
+You can now build and run your environment with:
+
+```shell
+docker compose up --build
+```
+
+The behavior of this WordPress setup should be the same as in the previous examples, but this time, the image is self-contained and doesn't rely on volumes set up within the host machine to work, in addition to not allowing new package installations or login through a shell.
+
+To stop the services, type `CTRL+C` in the terminal where the services are running, and then run:
+
+```bash
+docker-compose down
+```
+Please notice that although this setup allows for customizations through the WordPress dashboard, changes won't be persisted through container rebuilds.
```

Comment:
Rather than leave this "hole" would it be better to add:
```
services:
  app:
    environment:
      WORDPRESS_CONFIG_EXTRA |
        define('DISALLOW_FILE_MODS',true);
```
In the compose file? Then it disables most changes that are not persisted in the database.

Reply:
These are great suggestions! I think it makes total sense for the distroless setup. I will update the guide to include these and add notes about updates